src/child.cpp

```diff
@@ -0,0 +1,473 @@
+                        errc - error correction test\n\
+                        jb   - jitter/asymmetry test\n\
+                        ft   - focus/tracking test\n\
+                        ta   - time analyser\n\
+                speed - set test speed, VAL should be integer\n\
+                simul - set test speed, VAL should be 0 or 1 (default: 1)\n\
+";
+
+const int helpstr_sz = sizeof(helpstr);
+
+enum qscan_mode { none = 0, scanbus = 1, scan, dinfo, minfo, help };
+
+char tchar = -1;
+
+enum fdtype_t { FD_PIPE = 1, FD_SOCKET = 2 };
+
+// int readline(int fd, char *buf, int maxlen, fdtype_t fdtype = FD_PIPE)
+int readline(int fd, char *buf, int maxlen, fdtype_t fdtype) {
+  int cnt = 0;
+  char *cbuf = buf;
+  int r;
+  int sret;
+  fd_set rd_set;
+  timeval tv;
+
+  FD_ZERO(&rd_set);
+
+  //  printf("tchar=%02X\n", tchar);
+  if (tchar >= 0) {
+    cbuf[0] = tchar;
+    cnt++;
+    cbuf++;
+    tchar = -1;
+  }
+  while (!term && (cnt < (maxlen - 1))) {
+    FD_SET(fd, &rd_set);
+    tv.tv_sec = 1;
+    tv.tv_usec = 0;
+#ifdef _WIN32
+    if (fdtype == FD_SOCKET) {
+#endif
+      sret = select(fd + 1, &rd_set, NULL, NULL, &tv);
+#ifdef _WIN32
+      errno = GetLastError();
+    } else {
+      sret = 1;
+      errno = 0;
+    }
+#endif
+    //      printf("select(%d): %s\n", fd, strerror(errno));
+    if (sret < 0) {
+#ifndef _WIN32
+      if (debug) {
+        if (debug && !daemonized)
+          std::cout << "readline() " << fd << " select: " << strerror(errno)
+                    << "\n";
+      }
+#endif
+      if (errno == EINTR)
+        continue;
+      return -1;
+    } else if (sret > 0 && FD_ISSET(fd, &rd_set)) {
+#ifdef _WIN32
+      if (fdtype == FD_SOCKET)
+        r = recv(fd, cbuf, 1, 0);
+      else
+#endif
+        r = read(fd, cbuf, 1);
+      if (r < 0) {
+        if (debug && !daemonized)
+          std::cout << "read = " << r << ", " << errno << ", "
+                    << strerror(errno) << "\n";
+        switch (errno) {
+        case EAGAIN:
+          //  printf("EAGAIN\n");
+          continue;
+        case EINTR:
+          //  printf("EINTR\n");
+          continue;
+        default:
+          return -1;
+        }
+      }
+      if (!r)
+        return -1;
+      // look for CR/LF/CR+LF
+      if (fdtype == FD_SOCKET) {
+        if (buf[cnt] == 0x0A || buf[cnt] == 0x0D)
+          goto readline_end;
+      } else if (cnt && (buf[cnt - 1] == 0x0A || buf[cnt - 1] == 0x0D)) {
+        if (buf[cnt] != 0x0A && buf[cnt] != 0x0D)
+          tchar = buf[cnt];
+        buf[cnt - 1] = '\n';
+        buf[cnt] = 0;
+        return cnt;
+      }
+      cnt++;
+      cbuf++;
+    }
+  }
+  if (term)
+    return -1;
+readline_end:
+  buf[cnt] = '\n';
+  buf[cnt + 1] = 0;
+  return cnt + 1;
+}
+
+void child_proc(child_arg_t *arg) {
+  char speeds[16];
+  char linei[MAXLINE + 1];
+  char lineo[MAXLINE + IDENT_LEN + 1];
+  pipe_t pipefd;
+  pid_t cpid;
+  qscan_mode mode = none;
```

Comment:
Codacy found an issue: [Variable 'mode' is assigned a value that is never used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)